raiden/raiden_service.py

```diff
@@ -179,21 +198,49 @@ def __init__(self, chain, private_key_bin, transport, discovery, config):
 
         alarm.start()
 
-        if config['max_unresponsive_time'] > 0:
-            self.healthcheck = HealthcheckTask(
-                self,
-                config['send_ping_time'],
-                config['max_unresponsive_time']
-            )
-            self.healthcheck.start()
-        else:
-            self.healthcheck = None
-
         self.transaction_log = StateChangeLog(
             storage_instance=StateChangeLogSQLiteBackend(
                 database_path=config['database_path']
             )
         )
+
+        self.channels_serialization_path = None
+        self.channels_queue_path = None
+        if config['database_path'] != ':memory:':
+            self.channels_serialization_path = path.join(
+                path.dirname(self.config['database_path']),
+                'channels.pickle',
+            )
+
+            self.channels_queue_path = path.join(
+                path.dirname(self.config['database_path']),
+                'queues.pickle',
+            )
+
+            if path.exists(self.channels_serialization_path):
+                restored_channels = list()
+
+                with open(self.channels_serialization_path, 'r') as handler:
+                    try:
+                        while True:
+                            restored_channels.append(pickle.load(handler))
+                    except EOFError:
+                        pass
+
+                map(self.restore_channel, restored_channels)
```

Comment:
Changed to a for loop, didn't use a list comprehension because the output is not of interest (but this also applies to the `map` :P)